perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1928,6 +1959,37 @@ public Application getApplicationById(PerunSession sess, int appId) throws Regis
 
 	}
 
+	@Override
+	public List<Application> getApplicationsForGroup(PerunSession userSession, Group group, List<String> state, String dateFrom, String dateTo) throws PerunException {
+		groupsManager.checkGroupExists(userSession, group);
+
+		//Authorization
+		if (!AuthzResolver.authorizedInternal(userSession, "getApplicationsForGroup_Vo_List<String>_policy", Collections.singletonList(group))) {
+			throw new PrivilegeException(userSession, "getApplicationsForGroup");
+		}
+		if (state == null || state.isEmpty()) {
+			// list all
+			try {
+				return jdbc.query(APP_SELECT + " where a.group_id=? and a.created_at::date >= date ? and a.create_at::date <= date ? order by a.id desc", APP_MAPPER, group.getId(), dateFrom, dateTo);
+			} catch (EmptyResultDataAccessException ex) {
+				return new ArrayList<>();
+			}
+		} else {
+			// filter by state
+			try {
+				MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
+				sqlParameterSource.addValue("groupId", group.getId());
+				sqlParameterSource.addValue("states", state);
+				sqlParameterSource.addValue("from", dateFrom);
+				sqlParameterSource.addValue("to", dateTo);
+				return namedJdbc.query(APP_SELECT + " where a.group_id=:groupId and state in ( :states ) and a.created_at::date >= date :from and a.create_at::date <= date :to order by a.id desc", sqlParameterSource, APP_MAPPER);
```

Comment:
This SQL also seems to be incorrect. This will again work only if you use LocalDates.
```suggestion
				return namedJdbc.query(APP_SELECT + " where a.group_id=:groupId and state in ( :states ) and a.created_at::date >= :from and a.created_at::date <= :to order by a.id desc", sqlParameterSource, APP_MAPPER);
```

Reply:
See above.

Reply:
Changed to 'date(:param)'